src/transaction.rs

```diff
@@ -1,78 +1,360 @@
-use crate::nullifier::Nullifier;
-use crate::circuit::validity_predicate::ValidityPredicate;
-use crate::el_gamal::EncryptedNote;
-use crate::{
-    action::Action, add_to_tree, is_in_tree, note::Note, serializable_to_vec, CircuitParameters,
+use crate::action::Action;
+use crate::circuit::action_circuit::ActionCircuit;
+use crate::circuit::blinding_circuit::BlindingCircuit;
+use crate::circuit::circuit_parameters::CircuitParameters;
+use crate::circuit::validity_predicate::{ValidityPredicate, NUM_NOTE};
+use crate::constant::{
+    ACTION_CIRCUIT_SIZE, ACTION_PUBLIC_INPUT_CM_INDEX, ACTION_PUBLIC_INPUT_NF_INDEX,
+    ACTION_PUBLIC_INPUT_ROOT_INDEX, BLINDING_CIRCUIT_SIZE,
 };
-use ark_ec::twisted_edwards_extended::GroupAffine as TEGroupAffine;
-use rs_merkle::{algorithms::Blake2s, MerkleTree};
-
-pub struct Transaction<'a, CP: CircuitParameters> {
-    //max: usize, // the maximum number of actions/notes for a transaction
-    actions: Vec<Action<CP>>,
-    spent_notes: Vec<(Note<CP>, Nullifier<CP>)>,
-    created_notes: Vec<(Note<CP>, EncryptedNote<CP::InnerCurve>)>,
-    vps: &'a Vec<ValidityPredicate<CP>>,
+use crate::error::TaigaError;
+use crate::vp_description::ValidityPredicateDescription;
+use plonk_core::circuit::Circuit;
+use plonk_core::circuit::{verify_proof, VerifierData};
+use plonk_core::proof_system::{pi::PublicInputs, Proof, VerifierKey};
+use rand::RngCore;
+
+pub const NUM_TX_SLICE: usize = NUM_NOTE;
+
+pub struct Transaction<CP: CircuitParameters> {
```

Comment:
A transaction contains:
* one action per {spend+create} corresponding to the binding of the vks, etc.
* one spend check including the sender proof and the token proof
* one output check (is the name accurate?maybe creation_slice?) for the receiver proof and the token proof.